src/language-markdown/mdx.js

```diff
@@ -0,0 +1,64 @@
+"use strict";
+
+/**
+ * modified from https://github.com/mdx-js/mdx/blob/master/packages/mdx
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Compositor and Zeit, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+const IMPORT_REGEX = /^import/;
+const EXPORT_REGEX = /^export/;
+const BLOCKS_REGEX = "[a-z\\.]+(\\.){0,1}[a-z\\.]";
+const EMPTY_NEWLINE = "\n\n";
+
+const isImport = text => IMPORT_REGEX.test(text);
+const isExport = text => EXPORT_REGEX.test(text);
+
+const tokenizeEsSyntax = (eat, value) => {
+  const index = value.indexOf(EMPTY_NEWLINE);
+  const subvalue = value.slice(0, index);
+
+  if (isExport(subvalue) || isImport(subvalue)) {
```

Comment:
It's from [mdx-js/mdx](https://github.com/mdx-js/mdx/blob/30a740d3e7bfb71119f67998990239d3761cc4a9/packages/mdx/index.js#L28), IMO we shouldn't touch them if possible so that we could keep sync easily.